internal/ansible/ansible.go

```diff
@@ -202,15 +215,68 @@ func (r *Runner) handleJobEvent(event notify.EventInfo) {
 			ansibleEvent["end_line"] = 0
 		}
 
-		modifiedData, err := json.Marshal(ansibleEvent)
+		filteredEvent := filterEvent(ansibleEvent, r.schema)
+
+		modifiedData, err := json.Marshal(filteredEvent)
 		if err != nil {
 			log.Errorf("cannot marshal JSON: err=%v", err)
 			return
 		}
 
 		r.Events <- modifiedData
-		log.Debugf("event sent: event=%v", ansibleEvent)
+		log.Debugf("event sent: event=%v", prettyJson(filteredEvent))
+	}
+}
+
+// filterEvent filters an ansible job event `event` based on schema `schema`.
+//
+//	The function calls itself recursively to filter nested objects.
+//	All of this is loosely typed with map[string]any since we are dealing with
+//	job event JSON with nondeterministic properties.
+func filterEvent(event map[string]any, schema map[string]any) map[string]any {
+	properties, ok := schema["properties"]
+	if !ok {
+		properties = map[string]any{}
+	}
+
+	filteredEvent := map[string]any{}
+
+	for key, value := range event {
+		propSchema := properties.(map[string]any)[key]
```

Comment:
I would be careful before accessing anything in properties like this. You should double check what type properties has using something like this:

```go
var propSchema any
if reflect.TypeOf(properties).Kind() == reflect.Map {
	propSchema = properties.(map[string]any)[key]
}
```